src/tools/immutable_object_cache/CacheController.cc

```diff
@@ -64,10 +67,15 @@ void CacheController::handle_signal(int signum) {
   shutdown();
 }
 
-void CacheController::run() {
+int CacheController::run() {
   try {
     std::string controller_path =
       m_cct->_conf.get_val<std::string>("immutable_object_cache_sock");
+    if (controller_path.empty()) {
```

Comment:
Hi Jason, I want to describe such a scenario: if using default "/var/run/ceph/immutable_object_cache_sock", the root user had started the daemon, then "immutable_object_cache_sock" belongs to root. other users also use the default immutable_object_cache_sock path, they won't have permission . Or specify a immutable_object_cache_sock path that already exists and belongs to root. At this time, there should be a permission problem. Will the code run here and throw an error to tell users? The immutable_object_cache_path is also similar. So, should we suggest users to set a new path that the current user has permission to?

Reply:
> So, should we suggest users to set a new path that the current user has permission to?

Yes.

Reply:
woo,I finally found the problem I encountered